pages/in/widget/pay/index.vue

```diff
@@ -439,40 +440,50 @@ export default {
           paymentRedirectWhiteList,
         };
       });
+      let redirectWhitelistError = null;
+      let isRedirectURLWhiteListed = false;
       if (agentUser && !agentUser.cosmosWallet) {
-        error({ statusCode: 400, message: 'VIA_USER_HAS_NO_WALLET' });
+        redirectWhitelistError = redirectWhitelistError || { statusCode: 400, message: 'VIA_USER_HAS_NO_WALLET' };
       }
-
       if (toUsers.some(u => !u.cosmosWallet)) {
-        error({ statusCode: 400, message: 'RECEIPIENT_HAS_NO_WALLET' });
+        redirectWhitelistError = redirectWhitelistError || { statusCode: 400, message: 'RECEIPIENT_HAS_NO_WALLET' };
       }
-
       if (redirectUri) {
         if (agentId) {
           const {
             paymentRedirectWhiteList: agentWhiteList = [],
           } = agentUser;
           if (!IS_TESTNET && (!agentWhiteList || !agentWhiteList.length)) {
-            error({ statusCode: 400, message: 'AGENT_NOT_SETUP_PAYMENT_REDIRECT' });
+            redirectWhitelistError = redirectWhitelistError || { statusCode: 400, message: 'AGENT_NOT_SETUP_PAYMENT_REDIRECT' };
           }
           if (agentWhiteList.length && !agentWhiteList.includes(redirectUri)) {
-            error({ statusCode: 400, message: 'REDIRECT_URI_NOT_WHITELIST' });
+            redirectWhitelistError = redirectWhitelistError || { statusCode: 400, message: 'REDIRECT_URI_NOT_WHITELIST' };
+          }
+          if (agentWhiteList.length && agentWhiteList.includes(redirectUri)) {
+            isRedirectURLWhiteListed = true;
           }
         } else {
           if (toUsers.length > 1) {
-            error({ statusCode: 400, message: 'CANNOT_REDIRECT_MULTIPLE_RECEIPIENTS' });
+            redirectWhitelistError = redirectWhitelistError || { statusCode: 400, message: 'CANNOT_REDIRECT_MULTIPLE_RECEIPIENTS' };
           }
           const {
             paymentRedirectWhiteList: userWhiteList = [],
           } = toUsers[0];
           if (!IS_TESTNET && (!userWhiteList || !userWhiteList.length)) {
-            error({ statusCode: 400, message: 'USER_NOT_SETUP_PAYMENT_REDIRECT' });
+            redirectWhitelistError = redirectWhitelistError || { statusCode: 400, message: 'USER_NOT_SETUP_PAYMENT_REDIRECT' };
           }
           if (userWhiteList.length && !userWhiteList.includes(redirectUri)) {
-            error({ statusCode: 400, message: 'REDIRECT_URI_NOT_WHITELIST' });
+            redirectWhitelistError = redirectWhitelistError || { statusCode: 400, message: 'REDIRECT_URI_NOT_WHITELIST' };
+          }
+          if (userWhiteList.length && userWhiteList.includes(redirectUri)) {
+            isRedirectURLWhiteListed = true;
           }
         }
```

Comment:
I think we can just write `isRedirectURLWhiteListed = !err` here, at the end of the `if (redirectUri)` block
and no need to explicitly check for `if (agentWhiteList.length && agentWhiteList.includes(redirectUri))` and `if (userWhiteList.length && userWhiteList.includes(redirectUri)`